fili-core/src/test/groovy/com/yahoo/bard/webservice/web/ResponseWriterSelectorSpec.groovy

```diff
@@ -0,0 +1,97 @@
+// Copyright 2017 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web
+
+import com.yahoo.bard.webservice.application.ObjectMappersSuite
+import com.yahoo.bard.webservice.config.SystemConfig
+import com.yahoo.bard.webservice.config.SystemConfigProvider
+import com.yahoo.bard.webservice.data.ResultSet
+import com.yahoo.bard.webservice.table.Column
+import com.yahoo.bard.webservice.util.Pagination
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.DateTime
+
+import spock.lang.Specification
+
+class ResponseWriterSelectorSpec extends Specification {
+
+    private static CsvResponseWriter csvResponseWriter
+    private static JsonResponseWriter jsonResponseWriter
+    private static JsonApiResponseWriter jsonApiResponseWriter
+    private static FiliResponseWriterSelector filiResponseWriterSelector
+    private static final DebugWriter debugWriter = new DebugWriter();
+
+    private static final ObjectMappersSuite MAPPERS = new ObjectMappersSuite()
+
+    SystemConfig systemConfig = SystemConfigProvider.getInstance()
+    static final List<String> TYPE_NAMES_LIST = [
+            "Json",
+            "CSV",
+            "JsonApi"
+    ]
+    static final List<ResponseFormatType> TYPE_LIST = [
+            ResponseFormatType.JSON,
+            ResponseFormatType.CSV,
+            ResponseFormatType.JSONAPI
+    ]
+    static final List<Class<ResponseWriter>> WRITER_LIST = [
+            JsonResponseWriter,
+            CsvResponseWriter,
+            JsonApiResponseWriter
+    ]
+
+
+    Set<Column> columns
+    ResponseData response
+    DateTime dateTime = new DateTime(1000L * 60 * 60 * 24 * 365 * 45)
+    DataApiRequest apiRequest = Mock(DataApiRequest)
+    ResultSet resultSet
+    ByteArrayOutputStream os = new ByteArrayOutputStream()
+    Pagination pagination
+    SimplifiedIntervalList volatileIntervals = new SimplifiedIntervalList();
+
+    static class DebugWriter implements ResponseWriter {
+        @Override
+        void write(final ApiRequest request, final ResponseData responseData, final OutputStream os)
+                throws IOException {
+
+        }
+    }
+
+    def "test selector selecting correct writer for #TypeNames"() {
```

Comment:
Should add the `Unroll` annotation. The `Unroll` annotation expands each test case into its own separate test (substituting whatever variables need to be substituted in the process).